pkg/scheduler/framework/session_plugins.go

```diff
@@ -155,6 +158,26 @@ func (ssn *Session) AddJobStarvingFns(name string, fn api.ValidateFn) {
 	ssn.jobStarvingFns[name] = fn
 }
 
+func (ssn *Session) AddInitCycleStateFn(name string, fn api.InitCycleStateFn) {
```

Comment:
https://github.com/JesseStutler/volcano/blob/6078b0aabd4d8a617efca1a38430e8c2cf757166/pkg/scheduler/framework/session.go#L112-L117

https://github.com/volcano-sh/volcano/pull/4152, In this PR, I added a cycleStateMap to the session to store the cycleState of different tasks. Since cycleState may not only be used in predicates plugins. I think there is no need to add a new extension point called InitCycleStateFn? I think it is very likely that this extension point will only be used for simulation in preempt action. Therefore we can do it this way: Before simulation starts, we can clear the cycleStateMap again, and then simulateXXXFn can directly get the cycleState from this cycleStateMap. There is no need to add so many new extension points, which will confuse users. What do you think?